versioning/BRM/brmshmimpl.cpp

```diff
@@ -232,7 +240,163 @@ void BRMShmImpl::destroy()
     if (!oldName.empty()) bi::shared_memory_object::remove(oldName.c_str());
 }
 
+BRMManagedShmImpl::BRMManagedShmImpl(unsigned key, off_t size, bool readOnly)
+    : BRMShmImplParent(key, size, readOnly)
+{
+    string keyName = ShmKeys::keyToName(fKey);
+    off_t curSize = 0;
+
+    for(uint32_t tries = 0; fSize == 0 && tries <= ShmCreateMaxRetries; ++tries)
+    {
+        try
+        {
+            auto* shmSegment = new boost::interprocess::managed_shared_memory(bi::open_only, keyName.c_str());
+            curSize = shmSegment->get_size();
+
+            if (curSize == 0)
+                throw bi::interprocess_exception("shared memory segment size is 0.");
```

Comment:
Looks like a possible memory leak.

Reply:
I think execution never gets here b/c line 253 throws if the size is 0 but I will add a delete nevertheless.